src/toast/io/observation_hdf_load.py

```
@@ -689,47 +559,56 @@ def load_hdf5_obs_meta(
                     else:
                         msg = f"metadata object group '{obj_name}' has class "
                         msg += f"{obj.attrs['class']}, but instantiated "
-                        msg += f"object does not have a load_hdf5() method"
+                        msg += "object does not have a load_hdf5() method"
```

Should this be an error instead of a warning? How does a situation like this arise? Does the class only implement `save_hdf5()` and not `load_hdf5()`? or did someone write additional information into the HDF5 after is was created?

---

I had originally wanted to guard against files where other code has added attributes / datasets.  But this code is only executed if the `class` attribute is found, which means it should be a toast compatible object.  I can raise an exception instead of just warning the user.